src/lib/handler/observable-delegates-message-handler.middleware.ts

```diff
@@ -31,11 +31,16 @@ export class ObservableDelegatesMessageHandlerMiddleware extends AbstractDelegat
     return this._wrapWithObservable(message)
       .pipe(
         /**
-         * Resolves the handler based on the specified resolver.
-         * Wraps the handler into an observable to be sure to respect the chain.
+         * Resolves the set of handlers based on the specified resolver.
+         * Wraps the handlers into an observable to be sure to respect the chain.
          */
-        map((currentMessage: T) => this._messageHandlerResolver.getHandler(currentMessage)),
-        concatMap((handler: (message: T) => any) => this._wrapWithObservable(handler(message))),
+        map((currentMessage: T) => this._messageHandlerResolver.getHandlers(currentMessage)),
+        concatMap((handlers) => from(handlers)
+          .pipe(
+            flatMap((handler: (message: T) => any) => this._wrapWithObservable(handler(message))),
+            toArray()
```

Comment:
Please, use the `concatMap` instead of the flat map.
Also, the `toArray` creates an array of results that will be passed to `next` callback. Is this intended to work so?